crates/sui-indexer/src/indexer_reader.rs

```diff
@@ -1012,8 +1012,9 @@ impl IndexerReader {
                 .await;
         } else {
             let main_where_clause = match filter {
-                EventFilter::Package(package_id) => {
-                    format!("package = '\\x{}'::bytea", package_id.to_hex())
+                EventFilter::All(fs) if fs.is_empty() => {
```

Comment:
(Adding support for `"All": []` to the indexer reader -- this was a rare case where `IndexerReader` didn't have a way to do something that the fullnode did).